src/argilla/client/feedback/training/base.py

```diff
@@ -219,24 +240,31 @@ class ArgillaTrainerSkeleton(ABC):
     def __init__(
         self,
         feedback_dataset: "FeedbackDataset",
-        task_mapping: TrainingTaskMappingForTextClassification,
+        task: Union[
+            TrainingTaskForTextClassification,
+            TrainingTaskForSupervisedFinetuning,
+            TrainingTaskForRewardModelling,
+            TrainingTaskForDirectPreferenceOptimization,
```

Comment:
I've seen these block a couple times already, can't we just define it as `TrainingTaskTypes = Union[TrainingTaskForTextClassification, TrainingTaskForSupervisedFinetuning, TrainingTaskForRewardModelling, TrainingTaskForDirectPreferenceOptimization]`?

Reply:
Perhaps the best play is just to have one common superclass of all of these that we can use, but I'll go with this approach for now